src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/producer/MessageProducerImplUnitTest.java

```diff
@@ -0,0 +1,96 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.producer;
+
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.model.ChipsKafkaMessage;
+import uk.gov.companieshouse.kafka.producer.CHKafkaProducer;
+import uk.gov.companieshouse.service.ServiceException;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class MessageProducerImplUnitTest {
+
+    private static final String APP_ID = "chips-rest-interfaces-consumer";
+    private static final int ATTEMPT = 4;
+    private static final String MESSAGE_ID = "abc";
+    private static final String DATA = "{subject: testing}";
+    private static final String CHIPS_REST_ENDPOINT = "http://nowhere:1234";
+    private static final String CREATED_AT = "1610543925";
+
+    @Mock
+    private ApplicationLogger logger;
+
+    @Mock
+    private AvroSerializer avroSerializer;
+
+    @Mock
+    private CHKafkaProducer producer;
+
+    @InjectMocks
+    private MessageProducerImpl messageProducerImpl;
+
+    @Test
+    void testSuccessfulWriteToTopic()
+            throws ServiceException, IOException, ExecutionException, InterruptedException {
```

Comment:
IOException isn't thrown from the test, it can be removed

Reply:
Removed